NeoML/src/TraditionalML/SparseFloatMatrix.cpp

```diff
@@ -108,6 +108,40 @@ CSparseFloatMatrix& CSparseFloatMatrix::operator = ( const CSparseFloatMatrix& v
 	return *this;
 }
 
+void CSparseFloatMatrix::GrowInRows( int newRowsBufferSize )
+{
+	assert( newRowsBufferSize > 0 );
+	if( newRowsBufferSize > body->RowsBufferSize ) {
+		int newBufferSize = max( body->RowsBufferSize * 3 / 2, newRowsBufferSize );
+		CPtr<CSparseFloatMatrixBody> newBodyPart = FINE_DEBUG_NEW CSparseFloatMatrixBody( 0, 0, 0, newBufferSize, 0 );
+
+		::memcpy( newBodyPart->Desc.PointerB, body->Desc.PointerB, body->Desc.Height * sizeof( int ) );
+		::memcpy( newBodyPart->Desc.PointerE, body->Desc.PointerE, body->Desc.Height * sizeof( float ) );
+
+		CSparseFloatMatrixBody* modifiableBody = body.CopyOnWrite();
```

Comment:
We already have a body (newBodyPart). Why do you get additional?

Reply:
It's not additional, it's current modifiable version. I need newBodyPart only to allocate new particular buffers and then deallocate the current old ones (that's why I swap them)